lib/browser/api/menu.js

```diff
@@ -5,262 +5,115 @@ const EventEmitter = require('events').EventEmitter
-        }
-        if (!checked) {
-          v8Util.setHiddenValue(group[0], 'checked', true)
-        }
+      // Ensure radio groups have at least one menu item seleted
+      for (let id in this.groupsMap) {
+        let found = this.groupsMap[id].find(item => item.checked) || null
+        if (!found) v8Util.setHiddenValue(this.groupsMap[id][0], 'checked', true)
       }
     }
   }
 }
 
+// create and show a popup
 Menu.prototype.popup = function (window, x, y, positioningItem) {
   let asyncPopup
+  let [newX, newY, newPosition, newWindow] = [x, y, positioningItem, window]
 
   // menu.popup(x, y, positioningItem)
-  if (window != null && (typeof window !== 'object' || window.constructor !== BrowserWindow)) {
-    // Shift.
-    positioningItem = y
-    y = x
-    x = window
-    window = null
+  if (!window) {
+    // shift over values
+    if (typeof window !== 'object' || window.constructor !== BrowserWindow) {
+      [newPosition, newY, newX, newWindow] = [y, x, window, null]
+    }
   }
 
   // menu.popup(window, {})
-  if (x != null && typeof x === 'object') {
-    const options = x
-    x = options.x
-    y = options.y
-    positioningItem = options.positioningItem
-    asyncPopup = options.async
+  if (x && typeof x === 'object') {
+    const opts = x
+    newX = opts.x
+    newY = opts.y
+    newPosition = opts.positioningItem
+    asyncPopup = opts.async
   }
 
-  // Default to showing in focused window.
-  if (window == null) window = BrowserWindow.getFocusedWindow()
-
-  // Default to showing under mouse location.
-  if (typeof x !== 'number') x = -1
-  if (typeof y !== 'number') y = -1
-
-  // Default to not highlighting any item.
-  if (typeof positioningItem !== 'number') positioningItem = -1
-
-  // Default to synchronous for backwards compatibility.
+  // set defaults
+  if (typeof x !== 'number') newX = -1
+  if (typeof y !== 'number') newY = -1
+  if (typeof positioningItem !== 'number') newPosition = -1
+  if (!window) newWindow = BrowserWindow.getFocusedWindow()
   if (typeof asyncPopup !== 'boolean') asyncPopup = false
 
-  this.popupAt(window, x, y, positioningItem, asyncPopup)
+  this.popupAt(newWindow, newX, newY, newPosition, asyncPopup)
 }
 
+// close an existing popup
 Menu.prototype.closePopup = function (window) {
-  if (window == null || window.constructor !== BrowserWindow) {
+  if (!window || window.constructor !== BrowserWindow) {
     window = BrowserWindow.getFocusedWindow()
   }
-  if (window != null) {
-    this.closePopupAt(window.id)
-  }
+  if (window) this.closePopupAt(window.id)
 }
 
+// return a MenuItem with the specified id
 Menu.prototype.getMenuItemById = function (id) {
   const items = this.items
 
   let found = items.find(item => item.id === id) || null
-  for (let i = 0, length = items.length; !found && i < length; i++) {
+  for (let i = 0; !found && i < items.length; i++) {
     if (items[i].submenu) {
       found = items[i].submenu.getMenuItemById(id)
     }
   }
   return found
```

Comment:
Use `.find` ?

Reply:
where? sorry for confusion haha i'm just unclear since i use `.find` on line #90 :)

Reply:
for `items` array?

Reply:
would `let found = items.find(item => item.id === id) || null` not be that?

Reply:
yeah, sorry I'm just a bit confused that there is a loop just right after `.find` call that seem to be doing similar thing.

Reply:
ah, so all MenuItems can have submenus, so what the loop is doing is calling the method recursively on submenus if they exist :)